vcls/magento2/default.vcl

```diff
@@ -0,0 +1,229 @@
+    new dir = goto.dns_director("{{ host }}:{{ port }}",
+                                probe = p1,
+                                first_byte_timeout = 10m);
+}
+
+sub vcl_recv {
+    set req.backend_hint = dir.backend();
+
+    if (req.method == "PURGE") {
+        if (client.ip !~ purge) {
+            return (synth(405, "Method not allowed"));
+        }
+        # To use the X-Pool header for purging varnish during automated deployments, make sure the X-Pool header
+        # has been added to the response in your backend server config. This is used, for example, by the
+        # capistrano-magento2 gem for purging old content from varnish during it's deploy routine.
+        if (!req.http.X-Magento-Tags-Pattern && !req.http.X-Pool) {
+            return (synth(400, "X-Magento-Tags-Pattern or X-Pool header required"));
+        }
+        if (req.http.X-Magento-Tags-Pattern) {
+            set req.http.X-Magento-Tags-Pattern = regsuball(req.http.X-Magento-Tags-Pattern, "\(\(\^\|\,\)", " ");
+            set req.http.X-Magento-Tags-Pattern = regsuball(req.http.X-Magento-Tags-Pattern, "\(\,\|\$\)\)", "");
+            set req.http.X-Magento-Tags-Pattern = regsuball(req.http.X-Magento-Tags-Pattern, "\|", ",");
+            ykey.purge_keys(req.http.X-Magento-Tags-Pattern);
+        }
+        if (req.http.X-Pool) {
+            set req.http.X-Pool = regsuball(req.http.X-Pool, "\(\(\^\|\,\)", " ");
+            set req.http.X-Pool = regsuball(req.http.X-Pool, "\(\,\|\$\)\)", "");
+            set req.http.X-Pool = regsuball(req.http.X-Pool, "\|", ",");
+            ykey.namespace("pool");
+            ykey.purge_keys(req.http.X-Pool);
+            ykey.namespace_reset();
+        }
+        return (synth(200, "Purged"));
+    }
+
+    if (req.method != "GET" &&
+        req.method != "HEAD" &&
+        req.method != "PUT" &&
+        req.method != "POST" &&
+        req.method != "TRACE" &&
+        req.method != "OPTIONS" &&
+        req.method != "DELETE") {
+          /* Non-RFC2616 or CONNECT which is weird. */
+          return (pipe);
+    }
+
+    # We only deal with GET and HEAD by default
+    if (req.method != "GET" && req.method != "HEAD") {
+        return (pass);
+    }
+
+    # Bypass customer, shopping cart, checkout
+    if (req.url ~ "/customer" || req.url ~ "/checkout") {
+        return (pass);
+    }
+
+    # Bypass health check requests
+    if (req.url ~ "^/(pub/)?(health_check.php)$") {
+        return (pass);
+    }
+
+    # if the backend is healthy, limit the grace
+    if (std.healthy(req.backend_hint)) {
+        set req.grace = /* {{ grace_period }} */s;
+        set req.http.grace = "normal (healthy server)";
+    } else {
+        set req.http.grace = "unlimited (unhealthy server)";
+    }
+
+    set req.http.grace = "none";
+
+    # normalize url in case of leading HTTP scheme and domain
+    set req.url = regsub(req.url, "^http[s]?://", "");
+
+    # collect all cookies
+    std.collect(req.http.Cookie);
```

Comment:
No browser requires this.

Reply:
it doesn't hurt and is from upstream, leaving it